revscoring/utilities/union_merge_observations.py

```diff
@@ -0,0 +1,62 @@
+"""
+``revscoring union_merge_observations -h``
+::
+
+    Merge labeled revisions, taking the union of values for any rows with
+    the same id.  Data is triaged according to the order of filenames in the
+    commandline arguments, with the later files taking preference over earlier.
+    Behavior is not specified if an input file has duplicate revisions within
+    it.
+
+    FIXME: Reading everything into memory is reckless.  Estimate where this
+    hits a wall.
+
+    Usage:
+        union_merge_observations -h | --help
+        union_merge_observations <input>...
+            [--output=<path>]
+            [--id-column=<str>]
+
+    Options:
+        <input>           List of input file paths
+        --output=<path>   Path to write out the merged observations
+                          [default: <stdout>]
+        --id-column=<str> Name of the id column for deduplication.
+                          [default: rev_id]
+"""
+
+import collections
+import docopt
+import sys
+
+from .util import read_observations, dump_observation
+
+
+def main(argv=None):
+    args = docopt.docopt(__doc__, argv=argv)
+
+    if args['--output'] == "<stdout>":
+        out_file = sys.stdout
+    else:
+        out_file = open(args['--output'], "w")
+
+    observations = []
+    for path in args['<input>']:
+        with open(path, "r") as in_file:
+            observations.extend(read_observations(in_file))
```

Comment:
Yeah, it's unfortunate, and the entire output gets read into memory as well.  I made a note about this inefficiency in the commit message and the module's docstring, to do it later.  Might as well put a quick note here as well, since people will run into it and wonder as you've demonstrated.

Do you think we have to worry about it now, though?  I was going to wait until we run into the wall.  Estimating, editquality labeled revisions files are 180 bytes per row or 180MB per 1M observations.  A few of those files, Python overhead, and the worst case scenario of no duplicates would nary add up to a gigabyte.  If we start talking feature cache... but otherwise nah let's kick this down the road.

*How* to solve the problem is a cool algorithm question, though.  I don't know!  I can't think of anything clever off-hand, the tradeoff looks like disk access vs. memory.